app/src/org/commcare/connect/ConnectJobHelper.kt

```diff
@@ -150,20 +151,17 @@ object ConnectJobHelper {
                 ConnectJobUtils.storePayment(context, payment)
                 FirebaseAnalyticsUtil.reportCccApiPaymentConfirmation(true)
                 listener.connectActivityComplete(true)
+                listenerWithMsg?.connectActivityCompleteWithMsg(true, null)
             }
 
             override fun onFailure(
                 errorCode: PersonalIdOrConnectApiErrorCodes,
                 t: Throwable?,
             ) {
-                Toast
-                    .makeText(
-                        context,
-                        PersonalIdOrConnectApiErrorHandler.handle(context, errorCode, t),
-                        Toast.LENGTH_LONG,
-                    ).show()
+                val msg = PersonalIdOrConnectApiErrorHandler.handle(context, errorCode, t)
                 FirebaseAnalyticsUtil.reportCccApiPaymentConfirmation(false)
                 listener.connectActivityComplete(false)
+                listenerWithMsg?.connectActivityCompleteWithMsg(false, msg)
```

Comment:
What if we refactor `ConnectActivityCompleteListener.connectActivityComplete()` and avoid having to call two different listeners here?